botocore/client.py

```diff
@@ -523,9 +526,15 @@ class Config(object):
         * User agent extra
 
     """
+
+    DEFAULT_TIMEOUT = 60
```

Comment:
Good point. Originally I was thinking that the client.py is now in the driver's seat anyway, so it seemed reasonable to have its own definition. Also the endpoint.py's DEFAULT_TIMEOUT will always be overridden and arguably should be removed, but I still kept it for backward compatibility (just in case some customers directly use the Endpoint class constructor).

But I have no problem at all to change the client.py to import DEFAULT_TIMEOUT from endpoint.py.

And all other suggestions are also good, I will incorporate them.